coalib/bearlib/aspects/Smell.py

```diff
@@ -294,21 +494,49 @@ class docs:
 @Smell.subaspect
 class Complexity:
     """
-    This aspect checks on the cyclomatic complexity of your code.
+    Complexity of a code based on different complexity metrics.
     """
     class docs:
         example = """
-        for (i=0; i<n; ++i){
-            for (i=0; i<n; ++i){
-                for (i=0; i<n; ++i){
-                    for (i=0; i<n; ++i){
-                        for (i=0; i<n; ++i){
-                            for (i=0; i<n; ++i){
-                                for (i=0; i<n; ++i){
-                                    for (i=0; i<n; ++i){
-                                        ...
-                                        //do something
-        ...
+        * McCabe's complexity
+        * Halstead complexity
+        * Elshof complexity
+        * Data complexity
+        etc...
+
+        Here is an example of complex code:
+        https://github.com/sympy/sympy/blob/master/sympy/solvers/solvers.py
+        """
+        example_language = 'English'
+        importance_reason = """
+        Complex programs are difficult to read and maintain. Reducing a code's
+        complexity improves its organization.
+        """
+        fix_suggestions = """
+        Implementing simple methods, avoiding too many branches, avoiding too
+        much multilevel inheritance etc... can fix this.
+        """
+
+
+@Complexity.subaspect
+class CylomaticComplexity:
+    """
+    Number of linearly independent paths through a program's source code.
+
+    The `Cyclomatic complexity
+    <https://wikipedia.org/wiki/Cyclomatic_complexity>`_  was developed by
+    Thomas J. McCabe in 1976 and it is based on a control flow representation
+    of the program.
+    """
+    class docs:
+        example = """
+        // The cyclomatic complexity of this program is 4.
+
+        int foo (int a, int b) {
+            if (a > 17 && b < 42 && a+b < 55) {
+                return 1;
+            }
+            return 2;
         }
         """
         example_language = 'C++'
```

Comment:
this is C++?

It isnt original "C", but `//` comments have been in C a long time.